### PR TITLE
libgphoto2: update udev rules version

### LINKED ISSUE #145544: libgphoto2 probably needs updating udev generation
We currently use version `175` and export it to `40-gphoto2.rules`

https://github.com/NixOS/nixpkgs/blob/26f5e6dd5f2d703d5c39d661540397ea119aa94f/pkgs/development/libraries/libgphoto2/default.nix#L40

The help page suggests there is version 201 and the file name should be `90-libgphoto2.rules`:

https://github.com/gphoto/libgphoto2/blob/4a0d9dcb1fffd226925da095684dfce2184fe430/packaging/generic/print-camera-list.c#L1417 ([modes should called versions](https://github.com/gphoto/libgphoto2/pull/743))

The change will also likely need a hwdb file according to:

https://github.com/gphoto/libgphoto2/blob/4a0d9dcb1fffd226925da095684dfce2184fe430/packaging/generic/print-camera-list.c#L508

This is apparently required to support cameras that use SCSI tunneling support:

https://github.com/gphoto/libgphoto2/blob/4a0d9dcb1fffd226925da095684dfce2184fe430/packaging/generic/print-camera-list.c#L744-L746

cc @jcumming (maintainer) @league ([introduced](https://github.com/NixOS/nixpkgs/commit/6eead52e1252413ea835fa1261bbf73ff569d23d) the udev file)

### PR DESCRIPTION
**Pull request by Artturin**
and add hwdb

https://github.com/gphoto/libgphoto2/blob/48d9bafe1fb9b8bd7485d554b1fec8a6a9bd3f71/packaging/generic/print-camera-list.c#L1407-L1416

paths from
https://github.com/gphoto/libgphoto2/blob/48d9bafe1fb9b8bd7485d554b1fec8a6a9bd3f71/packaging/generic/print-camera-list.c#L1426 and https://github.com/gphoto/libgphoto2/blob/48d9bafe1fb9b8bd7485d554b1fec8a6a9bd3f71/packaging/generic/print-camera-list.c#L1413

ref
https://github.com/archlinux/svntogit-packages/blob/acd9af4de4b07d4fcdb08142f7eb596ae75d81d2/trunk/PKGBUILD#L57 https://src.fedoraproject.org/rpms/libgphoto2/blob/rawhide/f/libgphoto2.spec#_105

Closes https://github.com/NixOS/nixpkgs/issues/145544

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by jtojnar — APPROVED** [review]
LGTM, but I have no way to test.

File: pkgs/development/libraries/libgphoto2/default.nix [hunk 1]
```diff
@@ -58,10 +58,13 @@ stdenv.mkDerivation rec {
...
-      mkdir -p $out/lib/udev/rules.d
+      mkdir -p $out/lib/udev/{rules.d,hwdb.d}
       ${executablePrefix}/lib/libgphoto2/print-camera-list \
-          udev-rules version 175 group camera \
-          >$out/lib/udev/rules.d/40-gphoto2.rules
+          udev-rules version 201 group camera \
+          >$out/lib/udev/rules.d/40-libgphoto2.rules
+      ${executablePrefix}/lib/libgphoto2/print-camera-list \
+          hwdb version 201 group camera \
+          >$out/lib/udev/hwdb.d/20-gphoto.conf
```
Comment (jtojnar) [inline pkgs/development/libraries/libgphoto2/default.nix:67]: Actually, both Fedora and arch use hwdb as file extension, not sure if that is important.

**Comment by Artturin** [comment]
https://man.archlinux.org/man/core/systemd/hwdb.7.en
> . hwdb files must have the extension .hwdb; other extensions are ignored.

File: pkgs/development/libraries/libgphoto2/default.nix (hunk 1 above)
Comment (Artturin) [reply pkgs/development/libraries/libgphoto2/default.nix]: https://man.archlinux.org/man/core/systemd/hwdb.7.en
> . hwdb files must have the extension .hwdb; other extensions are ignored.

Comment (jtojnar) [reply pkgs/development/libraries/libgphoto2/default.nix]: Would be nice to fix that upstream. Or even better, make the build system install these files. But autotools are annoying.

**Pull request merged**